rs/sns/governance/src/governance.rs

```diff
@@ -2120,6 +2135,27 @@ impl Governance {
                 self.perform_manage_dapp_canister_settings(manage_dapp_canister_settings)
                     .await
             }
+            Action::AdvanceSnsTargetVersion(_) => {
+                get_action_auxiliary(&self.proto.proposals, ProposalId { id: proposal_id })
+                    .and_then(|action_auxiliary| {
+                        action_auxiliary.unwrap_advance_sns_target_version_or_err()
+                    })
+                    .and_then(|new_target| {
+                        let cached_upgrade_steps = self
+                            .proto
+                            .get_cached_upgrade_steps_or_err()
+                            .map_err(|err| {
+                                GovernanceError::new_with_message(ErrorType::NotFound, err)
+                            })?;
+                        cached_upgrade_steps
+                            .validate_new_target_version(&new_target)
+                            .map_err(|err| {
+                                GovernanceError::new_with_message(ErrorType::InvalidProposal, err)
+                            })?;
+                        self.proto.target_version = Some(new_target);
```

Comment:
There is already an `advance_target_version` function, which may need to be modified slightly as right now it takes  a `AdvanceTargetVersionRequest` when it should just take a version, but probably we should call it instead of just setting the target version here

Reply:
We agreed that test-only `advance_target_version` function can be removed once we have the proposals to replace it in testing.